AddonArmTemplate/FullAzureMonitorMetricsProfile.json

```diff
@@ -52,8 +52,8 @@
     "clusterSubscriptionId": "[split(parameters('clusterResourceId'),'/')[2]]",
     "clusterResourceGroup": "[split(parameters('clusterResourceId'),'/')[4]]",
     "clusterName": "[split(parameters('clusterResourceId'),'/')[8]]",
-    "dceName": "[Concat('MSProm', '-', parameters('azureMonitorWorkspaceLocation'), '-', variables('clusterName'))]",
-    "dcrName": "[Concat('MSProm', '-', parameters('azureMonitorWorkspaceLocation'), '-', variables('clusterName'))]",
+    "dceName": "[substring(Concat('MSProm', '-', parameters('azureMonitorWorkspaceLocation'), '-', variables('clusterName')), 0, 44)]",
```

Comment:
also kaveesh - it appears it cannot end with anything other than a-z or 0-9 , so if the name ends with any other allowed chars (ex;- "-"), it can fail as well.. same with DCR as well. DCRA doesn't have any limitation (looking at our Ux) ?

Reply:
It doesn't appear like there is any simple way of achieving this in ARM tempaltes. maybe I can add a comment telling users about this? The main ARM template deployment would still fail with an error saying 'character not allowed' if the substring function shortens it to something other than an alphanumeric character at the end.

What do you think @vishiy ?

Reply:
ok sounds good. thanks kaveesh